### PR TITLE
Add llvm patch to fix ManagedStatic memleak

### PR DESCRIPTION
**Pull request by hewj03**
Patch is downloaded from https://reviews.llvm.org/D52425
It can fix memory leak of ManagedStatic.

### CODE REVIEW

File: common_clang.cpp [hunk 1]
```diff
@@ -87,7 +87,12 @@ static llvm::sys::Mutex lazyCCInitMutex;
 
 static llvm::ManagedStatic<llvm::sys::SmartMutex<true> > compileMutex;
 
-void CommonClangTerminate() { llvm::llvm_shutdown(); }
+void CommonClangTerminate() {
+  llvm::llvm_shutdown();
+#ifndef USE_PREBUILT_LLVM
```
Comment (fzou1) [inline common_clang.cpp:92]: why should we add this macro here?

Comment (hewj03) [reply common_clang.cpp:92]: deleteManagedStaticMutex doesn't exist in pre-built llvm.
It only exists when we use llvm source and apply 0001-Adding-llvm-deleteManagedStaticMutex.patch to llvm.

**Review by fzou1 — APPROVED** [review]

**Review by woshiyifei — APPROVED** [review]

**Comment by dkurt** [comment]
Hi! This PR breaks my OpenCL for Android build CI: https://github.com/dkurt/compute-runtime-android/actions/runs/251022900

```
/build_native_igc/IGC/llvm/src/projects/opencl-clang/common_clang.cpp: In function 'void CommonClangTerminate()':
/build_native_igc/IGC/llvm/src/projects/opencl-clang/common_clang.cpp:93:9: error: 'deleteManagedStaticMutex' is not a member of 'llvm'
   llvm::deleteManagedStaticMutex();
         ^~~~~~~~~~~~~~~~~~~~~~~~
/build_native_igc/IGC/llvm/src/projects/opencl-clang/common_clang.cpp:93:9: note: suggested alternative: 'ManagedStaticBase'
   llvm::deleteManagedStaticMutex();
         ^~~~~~~~~~~~~~~~~~~~~~~~
         ManagedStaticBase
IGC/llvm/build/src/projects/opencl-clang/CMakeFiles/opencl-clang.dir/build.make:118: recipe for target 'IGC/llvm/build/src/projects/opencl-clang/CMakeFiles/opencl-clang.dir/common_clang.cpp.o' failed
make[2]: *** [IGC/llvm/build/src/projects/opencl-clang/CMakeFiles/opencl-clang.dir/common_clang.cpp.o] Error 1
CMakeFiles/Makefile2:13684: recipe for target 'IGC/llvm/build/src/projects/opencl-clang/CMakeFiles/opencl-clang.dir/all' failed
make[1]: *** [IGC/llvm/build/src/projects/opencl-clang/CMakeFiles/opencl-clang.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by hewj03** [comment]
Hi @dkurt, this can be fixed by #173 on ocl-open-90 branch.

**Comment by dkurt** [comment]
@hewj03, thanks! Seems like the problem is solved but now we get OOM during [build](https://github.com/dkurt/compute-runtime-android/actions/runs/251022900):
```
[100%] Built target igc_dll
Error processing tar file(exit status 1): write /build/NATIVE/IGC/llvm/src/test/Transforms/DCE/int_sideeffect.ll: no space left on device
##[error]Process completed with exit code 1.
```

**Comment by hewj03** [comment]
Seems like disk is full

**Pull request merged**